sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java

```diff
@@ -411,6 +414,7 @@ private void validateTranslationFunction(TypeDescriptor<?> inputType,
           null);
     }
 
+    @SuppressWarnings("WeakerAccess")
```

Comment:
Why is this necessary?

Reply:
IDEA code analysis will recommend this be `package-private`.  Because it is a `public static class` with no `final` and the author opted for `protected` I was cautious and opted not to change the scope as perhaps there is an idea it might be extended one day. I wanted to suppress the static analysis warning though.

Reply:
IDEA code analysis will recommend this be `package-private`.  Because it is a `public static class` with no `final` and the author opted for `protected` for this method, I was cautious and opted not to change the scope as perhaps there is an idea it might be extended one day. I wanted to suppress the static analysis warning though.